### PR TITLE
[Codegen]: Extract module/errors into a shared file

### PR DESCRIPTION
**Pull request by tarunrajput**
## Summary

This PR reduces code duplication by extracting all the errors in the module/errors into a single parsers/errors.js file. All the errors must drop the corresponding Flow or Typescript token in the name and take an extra language parameter in the constructor. Also, rename the hasteModuleName parameter to nativeModuleName. 

Part of https://github.com/facebook/react-native/issues/34872

## Changelog

[Internal] [Changed] - Extract all the parsers errors in the module/errors into a single parsers/errors.js file

## Test Plan

run ```yarn jest react-native-codegen``` and check all test case passes.

image

### CODE REVIEW

**Comment by cipolleschi** [comment]
This PR should depend on this: #34874.
Plus, there are some JS errors in CI that should be addressed.

**Comment by cipolleschi** [comment]
We merged #34874 earlier today! :D 
Could you rebase once more on top of that, please? 🙏

**Comment by tarunrajput** [comment]
> We merged #34874 earlier today! :D Could you rebase once more on top of that, please? 🙏

Sure.

**Review by cipolleschi — APPROVED** [review]
Thank you so much for this work! :D 
This will enable further improvements!

File: packages/react-native-codegen/src/parsers/errors.js [hunk 1]
```diff
@@ -31,6 +33,413 @@ class ParserError extends Error {
   }
 }
 
+class MisnamedModuleInterfaceParserError extends ParserError {
+  constructor(nativeModuleName: string, id: $FlowFixMe, language: string) {
```
Comment (cipolleschi) [inline packages/react-native-codegen/src/parsers/errors.js:37]: In #34916, another contributor suggested to use a union type to describe the parsers. I really like this suggestion, and `Flow` will help users typing the errors if we type this correctly.

Ideally, we can do something like:
```ts
type ParserType = 'Flow' | 'TypeScript'

// ...

// update all the constructors
class MisnamedModuleInterfaceParserError extends ParserError {
  constructor(nativeModuleName: string, id: $FlowFixMe, language: ParserType) {
// ...
```

Nothing else in the logic will change, but the types will be more precise.

What do you think? Could you please update the PR? :pray:

**Review by cipolleschi — CHANGES_REQUESTED** [review]

File: packages/react-native-codegen/src/parsers/errors.js (hunk 1 above)
Comment (tarunrajput) [reply packages/react-native-codegen/src/parsers/errors.js:37]: I agree. I'll update the PR

**Review by cipolleschi — APPROVED** [review]
Love it! ❤️  Thank you so much for this change!

**Pull request closed without merging**